src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -37,6 +37,10 @@
     /** A pattern matching the symbol: "/". */
     private static final Pattern SLASH_PATTERN = Pattern.compile("[\\\\/]");
 
+    /** A pattern to find the string: "// violation". */
+    private static final Pattern VIOLATION_PATTERN = Pattern
+            .compile(".*[ ]*//[ ]*violation[ ]*");
```

Comment:
lets discuss this pattern
`.*[ ]*` - literally says "any characters followed by zero or more spaces", it can be simplified to `.*`
`//` - seems like it should be escaped 

i propose `.*\/\/\s*violation\s*$`

Reply:
done for now, this will need changes when we incorporate frequency feature as in https://github.com/checkstyle/checkstyle/pull/10182#discussion_r659023717